packages/unmock-core/src/service/parser.ts

```diff
@@ -28,7 +29,12 @@ export class ServiceParser implements IServiceParser {
         ? serviceFile.contents.toString("utf-8")
         : serviceFile.contents;
 
-    const schema = jsYaml.safeLoad(contents);
+    const { val: schema, errors } = loas3(jsYaml.safeLoad(contents));
```

Comment:
Might be easier for debugging to separate this into two steps as `safeLoad` can also throw an error, but for reasons not related to loas3. But not important now.

Reply:
Good idea. Also, I'm not sure if concatenating the messages with a newline is the best idea - it would be better to indicate which line numbers in the original document are responsible.

Reply:
Sidenote for a future PR - we'll need to edit the schema passed to `safeLoad` to handle custom tags. We could separate this to two steps in that PR IMO.